File: KtorApi-WhiskerWatch/src/main/kotlin/com/example/dto/ForumDto.kt
package com.example.dto

import kotlinx.serialization.Serializable

/**
 * Clase DTO Serializable para la creacion de foros
 */
@Serializable
data class ForumCreateDto(
    var mapsId: String,
    var listMessages: List<ForumMessagesCreateDto>
)

/**
 * Clase DTO Serializable para la creacion de mensajes
 */
@Serializable
data class ForumMessagesCreateDto(
    var username: String,
    var message: String,
    var created_At: String
)

File: Frontend-WhiskerWatch/src/app/models/forum/forum-dto/forum-dto.ts
export class ForumDto {
  public id: string;
  public mapsId: string;
  public listMessages: Array<ForumMessagesDto>;

  constructor(){
    this.id = '';
    this.mapsId = '';
    this.listMessages = [];
  }
}

export class ForumMessagesDto {
  public id: string;
  public username: string;
  public message: string;
  public created_At: Date;

  constructor() {
    this.id = '';
    this.username = '';
    this.message = '';
    this.created_At = new Date();
  }
}
